lib/board.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class Board
+    # player tokens
+    X_TOKEN = :X
+    O_TOKEN = :O
+    
+    attr_reader :board
+    
+    def initialize
+        @board = %w[1 2 3 4 5 6 7 8 9]
+    end
+    
+    private
+    
+    # This method is used to validate if an position is already taken by a player
+    # @param index
+    # @return boolean
+    def validate_index(index)
+        if @board[index] == X_TOKEN || @board[index] == O_TOKEN
+            false
+        else
+            true
+        end
+    end
+
+    # this method is used to check if the given element is a valid board position
+    # @param position
+    # @return boolean
+    def board_position(position)
+        if position.to_i >= 0 && position.to_i < 9
+            true
+        else
+            false
+        end
+    end
+    
+    public
+
+    # this method is used to print the board on the terminal
+    def display_board
```

Comment:
Unfortunately sing `puts`, `p` or printing outside of `main.rb` is forbidden by the requirements for this project. Please find a way to return the values from this method.

[`REQUIRED`]

Reply:
fixed